Source/JavaScriptCore/runtime/JSArrayBufferPrototype.cpp

```diff
@@ -293,10 +295,75 @@ JSC_DEFINE_HOST_FUNCTION(arrayBufferProtoFuncResize, (JSGlobalObject* globalObje
     return JSValue::encode(jsUndefined());
 }
 
-JSC_DEFINE_HOST_FUNCTION(arrayBufferProtoFuncTransfer, (JSGlobalObject* globalObject, CallFrame* callFrame))
+// https://tc39.es/proposal-arraybuffer-transfer/#sec-arraybuffercopyanddetach
+enum class CopyAndDetachMode {
+    PreserveResizability,
+    FixedLength
+};
+static JSArrayBuffer* arrayBufferCopyAndDetach(JSGlobalObject* globalObject, JSArrayBuffer* arrayBuffer, size_t newByteLength, CopyAndDetachMode mode)
 {
-    // https://tc39.es/proposal-resizablearraybuffer/#sec-arraybuffer.prototype.transfer
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    ASSERT(arrayBuffer->impl()->sharingMode() == ArrayBufferSharingMode::Default);
+    bool isResizable = arrayBuffer->isResizableOrGrowableShared();
+
+    if (UNLIKELY(arrayBuffer->impl()->isDetached())) {
+        throwVMTypeError(globalObject, scope, "Receiver is detached"_s);
+        return nullptr;
+    }
+
+    if (!isResizable && newByteLength == arrayBuffer->impl()->byteLength()) {
+        // We should just transfer!
+        ArrayBufferContents contents;
+        if (UNLIKELY(!arrayBuffer->impl()->transferTo(vm, contents))) {
+            throwVMRangeError(globalObject, scope, makeString("transfer failed"_s));
```

Comment:
I don't think you need `makeString` here? (And likewise for the other "transfer failed" cases.)

Also, the error messages would probably be nicer to read if we say "ArrayBuffer transfer" and "ArrayBuffer resize" instead of just "transfer" and "resize".

Reply:
Fixed.